### DISCUSSION TITLE
Option to add custom foreground scale for both zoe and any models

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#107] [General]
Hi would it be possible to add the option where you can put any number in the foreground fill so that someone can be able to experiment with either doing negative foreground or higher than 3

**Comment by nagadomi** [comment]
Currently, foreground-scale is not a continuous value, so it is not possible to select anything other than 0,1,2,3.
More selection values can be added, but I am not willing to add excessive values.

Regarding negative values,
I plan to add it, perhaps it will be useful for DepthAnything output.
DepthAnything output looks foreground scale larger than ZoeDepth output.

**Comment by nagadomi** [comment]
Added support for negative foreground-scale. (-3, -2, -1, 0, 1, 2, 3)

**Reply by kaelsonofkrypto** [reply]
Thanks you are the man, it seems on any large model that negative 2 on foreground looks good at depth 1.0. That seems to be the sweet spot. Going at a higher depth than 1.0 with negative 2 foreground causes a wobble artifact so I will stick with depth 1.0. 

I did experience a weird glitch earlier today I converted a 35 minute TV show with any large model and for the first 8 minutes it was in 3D but then somewhere around the 8th minute it became very flat like the depth stopped being added In the conversion. This happened prior to the update you just made. so I'm gonna test with this new update with the negative foreground and once the conversion finishes in eight hours I'll let you know if everything turned out good.

**Reply by kaelsonofkrypto** [reply]
update everything worked, looks good on any large. Thanks

**Reply by nagadomi** [reply]
negative-forground-scale scales small depth values to large values, so I guess it also amplify depth flickering.
Flicker-reduction may reduce that a bit, but I haven't tested it much.

**Reply by nagadomi** [reply]
Fixed a bug that the first few frames are incorrect when flicker reduction (--ema-normalize) is enabled.

**Reply by kaelsonofkrypto** [reply]
thanks so much, appreciate it
